stmhal/uart.c

```diff
@@ -347,6 +363,9 @@ STATIC bool uart_tx_wait(pyb_uart_obj_t *self, uint32_t timeout) {
 }
 
 STATIC HAL_StatusTypeDef uart_tx_data(pyb_uart_obj_t *self, uint8_t *data, uint16_t len) {
+    if (self->uart.Init.HwFlowCtl & UART_HWCONTROL_CTS) {
+        return HAL_UART_Transmit(&self->uart, data, len, self->timeout);
```

Comment:
I don't understand the need to use self->timeout here.  Can you please explain?  Maybe add a comment?